app/controllers/application_controller.rb

```diff
@@ -220,6 +220,11 @@ def render_error(msg, status)
     render_403
   end
 
+  def render_extensible(page_view = action_name)
+    @page = ::Pages::Manager.find_page(controller_name, action_name)
+    @page.view ? render(@page.view) : fail("page #{@page.name} has no view assigned")
```

Comment:
Exception seems to be better, yes. Also page can be nil, this should be handled correctly.